haystack/components/routers/zero_shot_text_router.py

```diff
@@ -0,0 +1,159 @@
+        }
+    })
+    ```
+    """
+
+    def __init__(
+        self,
+        labels: List[str],
+        multi_label: bool = False,
+        model: str = "MoritzLaurer/deberta-v3-base-zeroshot-v1.1-all-33",
+        device: Optional[ComponentDevice] = None,
+        token: Optional[Secret] = Secret.from_env_var("HF_API_TOKEN", strict=False),
+        pipeline_kwargs: Optional[Dict[str, Any]] = None,
+    ):
+        """
+        :param labels: The set of possible class labels to classify each sequence into. Can be a single label,
+            a string of comma-separated labels, or a list of labels.
+        :param multi_label: Whether or not multiple candidate labels can be true.
+            If False, the scores are normalized such that the sum of the label likelihoods for each sequence is 1.
+            If True, the labels are considered independent and probabilities are normalized for each candidate by
+            doing a softmax of the entailment score vs. the contradiction score.
+        :param model: The name or path of a Hugging Face model for zero-shot text classification.
+        :param device: The device on which the model is loaded. If `None`, the default device is automatically
+            selected. If a device/device map is specified in `pipeline_kwargs`, it overrides this parameter.
+        :param token: The API token used to download private models from Hugging Face.
+            If this parameter is set to `True`, the token generated when running
+            `transformers-cli login` (stored in ~/.huggingface) is used.
+        :param pipeline_kwargs: Dictionary containing keyword arguments used to initialize the
+            Hugging Face pipeline for zero shot text classification.
+        """
+        torch_and_transformers_import.check()
+
+        self.token = token
+        self.labels = labels
+        self.multi_label = multi_label
+        component.set_output_types(self, **{label: str for label in labels})
+
+        pipeline_kwargs = resolve_hf_pipeline_kwargs(
+            huggingface_pipeline_kwargs=pipeline_kwargs or {},
+            model=model,
+            task="zero-shot-classification",
+            supported_tasks=["zero-shot-classification"],
+            device=device,
+            token=token,
+        )
+        self.pipeline_kwargs = pipeline_kwargs
+        self.pipeline = None
+
+    def _get_telemetry_data(self) -> Dict[str, Any]:
+        """
+        Data that is sent to Posthog for usage analytics.
+        """
+        if isinstance(self.pipeline_kwargs["model"], str):
+            return {"model": self.pipeline_kwargs["model"]}
+        return {"model": f"[object of type {type(self.pipeline_kwargs['model'])}]"}
+
+    def warm_up(self):
+        if self.pipeline is None:
+            self.pipeline = pipeline(**self.pipeline_kwargs)
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Serialize this component to a dictionary.
```

Comment:
Sorry to be a nitpick here but we try to have these pydocs to be consistent. Just copy/paste from some other class here :-)

Reply:
Will do! This was made before the recent changes 😅